src/main/java/at/hannibal2/skyhanni/features/rift/area/mountaintop/SunGeckoHelper.kt

```diff
@@ -0,0 +1,291 @@
+
+    /**
+     * REGEX-TEST: §a[⬛⬛⬛⬜§c⬜ §e§lx2 §c⬜⬜⬜⬜⬜]
+     * REGEX-TEST: §c[⬜⬜⬜⬜⬜ §e§lx1 §c⬜⬜⬜⬜⬜]
+     * REGEX-TEST: §a[⬛⬛⬛⬛§c⬛ §e§lx1 §c⬛⬜⬜⬜⬜]
+     * REGEX-TEST: §a[⬛⬛⬛⬛⬛ §e§lx1 §a⬛§c⬛⬛⬜⬜]
+     * REGEX-TEST: §a[⬛⬛⬛⬛⬛ §e§lx1 §c§c⬛⬛⬜⬜⬜]
+     * REGEX-TEST: §a[⬛⬛⬛§c⬛⬛ §e§lx2 §c⬛⬛⬛⬜⬜]
+     */
+    private val sunGeckoActionBar by patternGroup.pattern(
+        "actionbar",
+        "(?<firstHalf>§[ac]\\[.*) §e§lx(?<combo>\\d+) (?<secondHalf>§[ac].*)]",
+    )
+
+    private var healthLeft = 250
+    private var totalHealth = 250
+    private var actionBarFormatted: String = ""
+    private var onFirstPhase = true
+    private var timeSliceDuration = Duration.ZERO
+    private var timeSinceLastHit = SimpleTimeMark.farPast()
+    private var combo = 1
+    private var scanningChat = false
+    private var inTimeChamber = false
+    private var currentBoss: Mob? = null
+
+    fun reset() {
+        display.clear()
+        healthLeft = 250
+        totalHealth = 250
+        actionBarFormatted = ""
+        onFirstPhase = true
+        timeSliceDuration = Duration.ZERO
+        timeSinceLastHit = SimpleTimeMark.farPast()
+        combo = 1
+        inTimeChamber = false
+        currentBoss = null
+    }
+
+    fun updateDisplay() {
+        display.clear()
+        var displayHealthLeft = healthLeft
+        var displayTotalHealth = totalHealth
+        if (modifiers.contains(MODIFIERS.REVIVAL)) {
+            if (onFirstPhase) displayHealthLeft += totalHealth
+            displayTotalHealth *= 2
+        }
+
+        var healthColor = "§a"
+        if (displayHealthLeft <= displayTotalHealth / 2) {
+            healthColor = "§e"
+        }
+
+        val health = "$healthColor$displayHealthLeft§f/§a$displayTotalHealth§c❤"
+        display.add("§eSun Gecko $health")
+        display.add("$actionBarFormatted §e§lCombo: x$combo")
+
+        // this is just a total guess but it looks right enough
+        // i think its inconsistent because of how often the action bar updates
+        var expiryTime = 5.seconds + 200.milliseconds
+        if (modifiers.contains(MODIFIERS.COLLECTIVE)) {
+            expiryTime += (modifiers.count() * 200).milliseconds
+        }
+        val timeLeft = timeSinceLastHit + expiryTime
+        if (timeLeft.timeUntil().inWholeMilliseconds > expiryTime.inWholeMilliseconds - 800.milliseconds.inWholeMilliseconds ||
+            timeLeft.isInPast()) {
+            display.add("§aCombo Timer: ${expiryTime.format()}/${expiryTime.format()}")
+        } else {
+            display.add("§aCombo Timer: ${timeLeft.timeUntil().format(showMilliSeconds = true)}/${expiryTime.format()}")
+        }
+
+        if (config.showModifiers) {
+            display.add("§6Modifiers:")
+            if (modifiers.isEmpty()) {
+                display.add("§eNone")
+            }
+            for (modifier in modifiers) {
+                display.add("§e${modifier.name.allLettersFirstUppercase()}")
```

Comment:
instead of calling .allLettersFirstUppercase() every time you update the display, make a variable in the enum called `displayname` that just does this instead when the enum is initialized